python/tests/avro/test_reader.py

```diff
@@ -20,7 +20,7 @@
 
 import pytest
 
-from pyiceberg.avro.decoder import BinaryDecoder
+from pyiceberg.avro.decoder import StreamingBinaryDecoder
```

Comment:
Should we also loop over `AVAILABLE_DECODERS` here?

Reply:
Added additional tests.